backend/src/repository/fhir_repository.py

```diff
@@ -102,19 +102,17 @@ def get_immunization_by_identifier(self, identifier_pk: str) -> tuple[Optional[d
         else:
             return None, None
 
-    def get_immunization_by_id(self, imms_id: str) -> Optional[dict]:
+    def get_immunization_by_id(self, imms_id: str) -> tuple[Optional[dict], Optional[str]]:
```

Comment:
I do think `tuple` is a less misleading return value than the previous `dict`, but I still had to read the last line to know what that tuple contained. Maybe something like `get_immunisation_and_version_by_id` would be clearer?